from_dict/_from_dict.py

```diff
@@ -286,8 +290,6 @@ def handle_dict_argument(
         if _get_constructor_type_hints(cls_arg_type_args[1]):
             # Dict[a,b]; we only support b being a structure.
             key_type, value_type = cls_arg_type_args
-            if fd_check_types:  # Perform type check on keys
```

Comment:
the `fd_check_types` parameter is no longer used here but I wanted to save that cleanup for another pull request.
I think It will make merging back into the branch with all the changes easier.